packages/astro/src/assets/fonts/vite-plugin-fonts.ts

```diff
@@ -107,16 +107,19 @@ export function fontsPlugin({ settings, sync, logger }: Options): Plugin {
 		hashToUrlMap = new Map();
 		resolvedMap = new Map();
 
+		const root = settings.config.root;
+
 		await loadFonts({
-			root: settings.config.root,
+			root,
 			base: baseUrl,
-			providers: settings.config.experimental.fonts!.providers ?? [],
-			// TS is not smart enough
-			families: settings.config.experimental.fonts!.families as any,
+			families: await Promise.all(
+				settings.config.experimental.fonts!.map((family: FontFamily<any>) =>
+					resolveFontFamily({ family, resolveMod, root }),
+				),
+			),
```

Comment:
I don't remember the code of  `resolveFontFamily`, so take this comment of mine with a grain of salt. I assume the resolution involves some I/O operation (read/write the FS and/or fetching something). If so, is there some caching in place?

Reply:
There's no IO, it internally resolves providers so errors are already handled there

Reply:
Yeah, one thing to understand is that "resolve" is a very generic term, that varies based on its context 😅 since I don't have much context, I don't understand what it means